Last-Update: 2014-03-28
Forwarded: not-needed
Origin: upstream, http://tracker.ceph.com/projects/ceph/repository/revisions/acc31e75a3e7115c00f9980609948455e3b2d49e/diff/
Bug-Ceph: http://tracker.ceph.com/issues/7849
Author: Sage Weil <sage@inktank.com>
Description: prevent 'ceph-conf' from making log files
    Subject: ceph-conf: do not log
    .
    If you are querying the conf for an osd and it has a log configured, we
    should not generate any log activity.
    .
    This isn't super pretty, but it is much less intrusive that wiring a 'do
    not log' flag down into CephContext and a zillion other places.

--- a/src/tools/ceph_conf.cc
+++ b/src/tools/ceph_conf.cc
@@ -157,8 +157,13 @@
   argv_to_vec(argc, argv, args);
   env_to_vec(args);
   vector<const char*> orig_args = args;
 
+  args.push_back("--log-file");
+  args.push_back("");
+  args.push_back("--no-log-to-stderr");
+  args.push_back("--no-log-to-syslog");
+
   global_init(NULL, args, CEPH_ENTITY_TYPE_CLIENT, CODE_ENVIRONMENT_DAEMON,
 	      CINIT_FLAG_NO_DAEMON_ACTIONS);
 
   // do not common_init_finish(); do not start threads; do not do any of thing
